modules/tensor_mechanics/test/tests/ad_elastic/tests

```diff
@@ -0,0 +1,234 @@
+  [../]
+  [./rz_incremental_small_elastic]
+    type = 'Exodiff'
+    input = 'rz_incremental_small_elastic.i'
+    exodiff = 'rz_incremental_small_elastic_out.e'
+    prereq = 'rz_incremental_small_elastic-noad'
+    requirement = 'We shall be able to reproduce incremental small strain elasticity results of the hand-coded simulation in cylindrical coordinates using automatic differentiation.'
+  [../]
+  [./rz_incremental_small_elastic-jac]
+    type = 'PetscJacobianTester'
+    input = 'rz_incremental_small_elastic.i'
+    run_sim = 'True'
+    prereq = 'rz_incremental_small_elastic'
+    ratio_tol = 3e-2
+    difference_tol = 1e0
+    cli_args = '-snes_test_err 1e-9 -mat_fd_type ds'
+    requirement = 'The Jacobian for the AD incremental small strain elasticity problem in cylindrical coordinates shall be perfect'
+  [../]
+
+  [./rz_small_elastic-noad]
+    type = 'Exodiff'
+    input = 'rz_small_elastic-noad.i'
+    exodiff = 'rz_small_elastic_out.e'
+    requirement = 'We shall be able to reproduce small strain elasticity results of the hand-coded simulation in cylindrical coordinates using automatic differentiation. (non-AD reference)'
+  [../]
+  [./rz_small_elastic]
+    type = 'Exodiff'
+    input = 'rz_small_elastic.i'
+    exodiff = 'rz_small_elastic_out.e'
+    prereq = 'rz_small_elastic-noad'
+    requirement = 'We shall be able to reproduce small strain elasticity results of the hand-coded simulation in cylindrical coordinates using automatic differentiation.'
+  [../]
+  [./rz_small_elastic-jac]
+    type = 'PetscJacobianTester'
+    input = 'rz_small_elastic.i'
+    run_sim = 'True'
+    prereq = 'rz_small_elastic'
+    ratio_tol = 3e-2
+    difference_tol = 1e0
+    cli_args = '-snes_test_err 1e-9 -mat_fd_type ds'
+    requirement = 'The Jacobian for the AD small strain elasticity problem in cylindrical coordinates shall be perfect'
+  [../]
+
+  #
+  # RSpherical 1D
+  #
+
+  [./rspherical_finite_elastic-noad]
+    type = 'Exodiff'
+    input = 'rspherical_finite_elastic-noad.i'
+    exodiff = 'rspherical_finite_elastic_out.e'
+    requirement = 'We shall be able to reproduce finite strain elasticity results of the hand-coded simulation in spherical coordinates using automatic differentiation. (non-AD reference)'
+  [../]
+  [./rspherical_finite_elastic]
+    type = 'Exodiff'
+    input = 'rspherical_finite_elastic.i'
+    exodiff = 'rspherical_finite_elastic_out.e'
+    prereq = 'rspherical_finite_elastic-noad'
+    requirement = 'We shall be able to reproduce finite strain elasticity results of the hand-coded simulation in spherical coordinates using automatic differentiation.'
+  [../]
+  [./rspherical_finite_elastic-jac]
+    type = 'PetscJacobianTester'
+    input = 'rspherical_finite_elastic.i'
+    run_sim = 'True'
+    prereq = 'rspherical_finite_elastic'
+    ratio_tol = 7e-2
```

Comment:
Yea, there is definitely something wrong here...8 iterations with Newton, only 5 with PJFNK

Reply:
It's definitely coming from displacement effects in the kernel because if I set `use_displaced_mesh = false`, then the Jacobian is perfect. So it appears to be my problem :smile:

Reply:
Ok, actually I'm going to say the problem is you and me :smile: So the error comes from:
<pre>
  return _grad_test[_i][_qp](0) * _stress[_qp](0, 0) +               // stress_{rr} part 1
         +(_test[_i][_qp] / <b>_q_point[_qp](0))</b> * _stress[_qp](1, 1) + // stress_{\theta \theta}
         +(_test[_i][_qp] / <b>_q_point[_qp](0))</b> * _stress[_qp](2, 2);  // stress_{\phi \phi}
</pre>
The location of the quadrature point actually changes with the displacements! So what we will need is the volume equivalent of `Assembly::_ad_q_points_face`, e.g. `Assembly::_ad_q_points`. I will get that coded up early next week unless you want to do it

Reply:
So the Jacobian difference was not just due to the `_q_point`. If I comment out that code, I still get a 10% Jacobian difference...I'm going to have to investigate more...